### ISSUE TITLE
Flaky test: RequestBody_ReadAsync_Success 

### ISSUE THREAD

**Issue by ryanbrandenburg** [aspnet/HttpSysServer#457]
This test [fails](http://aspnetci/viewLog.html?buildId=483224&buildTypeId=XPlat_Windows_Win2008r2_Universe) occasionally with the following error:
```
System.TimeoutException : AcceptAsync has timed out.
   at Microsoft.AspNetCore.Server.HttpSys.Listener.Utilities.AcceptAsync(HttpSysListener server, TimeSpan timeout) in /_/test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/Utilities.cs:line 108
   at Microsoft.AspNetCore.Server.HttpSys.Listener.RequestBodyTests.RequestBody_ReadAsync_Success() in /_/test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/RequestBodyTests.cs:line 73
--- End of stack trace from previous location where exception was thrown ---
```

**Comment by Tratcher** [comment]
History:
http://aspnetci/project.html?projectId=XPlat&buildTypeId=&tab=testDetails&testNameId=-5284826571007502406&order=TEST_STATUS_DESC&branch_XPlat=__all_branches__&itemsCount=50

18 failures since Feb 6, all on 2008R2 - Possible agent issue?

**Comment by ryanbrandenburg** [comment]
What would the issue on the agent be that would cause this behavior?

**Comment by Tratcher** [comment]
I don't know but it's suspicious that this only fails on 2008R2 and not Win7, they're the same OS. And we only have one agent for 2008R2 so it's hard to rule out an agent issue.

**Comment by ryanbrandenburg** [comment]
RequestBody_ReadAsync_Success [failed again](http://aspnetci/viewLog.html?buildId=516245&buildTypeId=XPlat_Windows_Win7_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
AuthTypes_AllowAnonymous_NoChallenge [failed again](http://aspnetci/viewLog.html?buildId=516731&buildTypeId=XPlat_Windows_Win7_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by Eilon** [comment]
@Tratcher - please take a look at this issue because it has been failing again.

**Comment by ryanbrandenburg** [comment]
AuthTypes_AllowAnonymous_NoChallenge [failed with about the same error](http://aspnetci/viewLog.html?buildId=521545&buildTypeId=XPlat_Windows_Win2008r2_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
RequestBody_ReadAsync_Success [failed with about the same error](http://aspnetci/viewLog.html?buildId=524632&buildTypeId=XPlat_Windows_Win2008r2_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
RequestBody_ReadAsync_Success [failed with about the same error](http://aspnetci/viewLog.html?buildId=529091&buildTypeId=XPlat_Windows_Win2008r2_Universe) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
RequestBody_ReadAsync_Success [failed with about the same error](http://aspnetci/viewLog.html?buildId=531393&buildTypeId=XPlat_Windows_Win2008r2_Universe) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
RequestBody_ReadAsync_Success [failed with about the same error](http://aspnetci/viewLog.html?buildId=537001&buildTypeId=XPlat_Windows_Win2008r2_Universe) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
RequestBody_ReadAsync_Success [failed with about the same error](http://aspnetci/viewLog.html?buildId=539955&buildTypeId=XPlat_Windows_Win2008r2_Universe) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
RequestBody_ReadAsync_Success [failed with about the same error](http://aspnetci/viewLog.html?buildId=548852&buildTypeId=XPlat_Windows_Win2008r2_Universe) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by Eilon** [comment]
@Tratcher any update on this? What are the next steps?

**Comment by Tratcher** [comment]
Recent failures all have this error:
```
System.Threading.Tasks.TaskCanceledException : A task was canceled.
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.AspNetCore.Server.HttpSys.Listener.RequestBodyTests.SendRequestAsync(String uri, HttpContent content) in /_/test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/RequestBodyTests.cs:line 361
   at Microsoft.AspNetCore.Server.HttpSys.Listener.Utilities.Before[T](Task`1 acceptTask, Task`1 responseTask) in /_/test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/Utilities.cs:line 122
   at Microsoft.AspNetCore.Server.HttpSys.Listener.RequestBodyTests.RequestBody_ReadAsync_Success() in /_/test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/RequestBodyTests.cs:line 73
--- End of stack trace from previous location where exception was thrown ---
```
The client appears to be hitting a 10s client side timeout. @JunTaoLuo this may be related to the failure you saw in https://github.com/aspnet/HttpSysServer/issues/468, except it's a lot more frequent. There was theory this was related to a client side bug but that hasn't been tracked down.

All failures are still on Win2008R2-Universe which implies the issue is OS or agent specific. No Win7 failures which should share the same OS components.

**Comment by aspnet-hello** [comment]
RequestBody_ReadAsync_Success [failed with about the same error](http://aspnetci/viewLog.html?buildId=554473&buildTypeId=XPlat_Windows_Win2008r2_Universe) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
There were 1 failures [with about the same error](http://aspnetci/viewLog.html?buildId=559384&buildTypeId=XPlat_Windows_Win7_Universe) on master:
- RequestBody_ReadAsync_Success


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact ryanbrandenburg.

There were 1 failures [with about the same error](http://aspnetci/viewLog.html?buildId=568037&buildTypeId=XPlat_Windows_Win2008r2_Universe) on master at 5:17:26 PM:
- RequestBody_ReadAsync_Success

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact ryanbrandenburg.

There were 1 failures [with about the same error](http://aspnetci/viewLog.html?buildId=570351&buildTypeId=XPlat_Windows_Win7_Universe) on master at 11:39:41 AM:
- RequestBody_ReadAsync_Success

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact ryanbrandenburg.

There were 1 failures [with about the same error](http://aspnetci/viewLog.html?buildId=570630&buildTypeId=XPlat_Windows_Win7_Universe) on master at 2:21:36 PM:
- RequestBody_ReadAsync_Success

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact ryanbrandenburg.

There were 1 failures [with about the same error](http://aspnetci/viewLog.html?buildId=574216&buildTypeId=XPlat_Windows_Win7_Universe) on master at 9:38:58 PM:
- RequestBody_ReadAsync_Success

**Comment by Eilon** [comment]
@Tratcher any news on this one?

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact ryanbrandenburg.

There were 1 failures [with about the same error](http://aspnetci/viewLog.html?buildId=577427&buildTypeId=XPlat_Windows_Win7_Universe) on master at 12:17:56 AM:
- RequestBody_ReadAsync_Success

**Comment by Tratcher** [comment]
Every indication here is that it's a client side SocketHttpHandler/Socket bug.
- https://github.com/aspnet/HttpSysServer/issues/451 includes a lot of similar failures across a variety of tests.
- Every failure is .NET Core only, no .NET 4.x failures.
- Every failure is Win7/2008R2 
- https://github.com/dotnet/corefx/issues/30691 may be related, though the error message is different.

I'm working on a local repro.

**Comment by Eilon** [comment]
BTW if it makes sense to temporarily disable this test, please do so (per the workflow at the top of this issue).

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact ryanbrandenburg.

There were 1 failures [with about the same error](http://aspnetci/viewLog.html?buildId=578273&buildTypeId=XPlat_Windows_Win7_Universe) on master at 12:01:04 PM:
- RequestBody_ReadAsync_Success

**Comment by Tratcher** [comment]
Update. I spent a whole day trying to reproduce these and did not get a single hit. I did come up with a new theory though. For historical purposes there are two versions of most of these tests, an ASP.NET level one and a lower level Listener version. Every failure reported here is in the Listener version of these tests.

This lower level doesn't have a message pump, it has to call Accept for each expected request and then switches back and forth between client and server code on the same thread. Example:
https://github.com/aspnet/HttpSysServer/blob/76cb2784b90b5549427700d4d93e44803c49101c/test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/AuthenticationTests.cs#L29-L40

This has a few issues.
- Readability...
- It has easily lead to deadlocks on single core machines.
- It is flakier under load or on slow machines like AppVeyor and apparently our CI.

New proposal: Remove the redundant tests. They are flaky and we would not lose any coverage. I'll give more history in the PR where I can show the changes.
